### PR TITLE
Close #373

### LINKED ISSUE #373: Set a Cloudlet to execute for a certain amount of TIME instead of MI
# REQUESTING A FEATURE:

## Detailed information about how the feature should work
When creating a cloudlet, set it to execute for a certain amount of TIME instead of MI(Million Instructions).

## An example scenario where this feature should be used
In PlanetLab example, we may want the cloudlet to finish as soon as the PlanetLab trace file finishes. But currently, the trace is replayed in a circular way until the cloudlet finishes all its MI.

## A brief explanation of why you think this feature is useful
It will be useful for people doing utilization-oriented VM scheduling simulation based on PlanetLab traces.

### PR DESCRIPTION
**Pull request by sun-lingyu**
# Close #373

#### What's this Pull Request does, clear and precisely?
This pull request added `lifeTime` attr and several related methods for a cloudlet. Users can use `setLifeTime()` method to set it, and when it is set, the cloudlet will finish execution after `lifeTime`  after its execution starts. 

#### Where should the reviewer start?
You can start from `CloudletSchedulerAbstract.java`.

#### Are there some simulation examples to show the performed changes working?
This pull request added a example(`PlanetLabExample2`)  to demonstrace its usage.

#### Any additional information you want to provide?
Currently lifetime can only be greater than `datacenter.SchedulingInterval`. Otherwise the cloudlet will finish after `datacenter.SchedulingInterval` has passed. Future work is needed to find out the reason and fix it.

### CODE REVIEW

**Comment by manoelcampos** [comment]
Helo @sun-lingyu 
Thanks for that.
I have to review it. Taking a quick look, there are some small things that need to be updated.
Since holidays are knocking on the door, I don't think we'll be able to complete the review and merge the changes before that. I'll make some small changes by now and try to provide some feedback.

Thanks for you contribution.

**Review by manoelcampos — CHANGES_REQUESTED** [review]
Hey @sun-lingyu,

I've made lots of small changes. Check the commits for details.
There are some other small issues I spotted in a quick review.
I won't be able to perform a new review round in the next days,
but I'll do it ASAP.

File: cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/traces/PlanetLabExample2.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+/**
+ * An example showing how to use the {@link UtilizationModelPlanetLab} class
+ * to define Cloudlets' CPU utilization based on <a href="https://www.planet-lab.org">PlanetLab's</a> trace files.
+ * Check the {@link #createCloudlets()} method.
+ * <p>It uses the new {@link Cloudlet#setLifeTime(double)} to
+ * define the maximum time the Cloudlet is allowed to execute.
+ * After this time is reached, the Cloudlet is finished as soon as possible.</p>
+ *
+ * @author Manoel Campos da Silva Filho
```
Comment (manoelcampos) [inline cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/traces/PlanetLabExample2.java:63]: You can add your name here

File: cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/traces/PlanetLabExample2.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+ * @since CloudSim Plus 7.2.1
+ */
+public class PlanetLabExample2 {
```
Comment (manoelcampos) [inline cloudsim-plus-examples/src/main/java/org/cloudsimplus/examples/traces/PlanetLabExample2.java:66]: The name of the class could be `PlanetLabCloudletLifeTimeExample`.
This numbering in examples were used in the beginning, but currently I'm trying to use more meaningful example names.

**Comment by sun-lingyu** [comment]
@manoelcampos 
Thank you very much!! 
As a beginner, I've really learnt a lot. 
Wish you a happy holiday:)

**Comment by manoelcampos** [comment]
Great. Happy holidays. I'll carry the review on soon.

**Review by manoelcampos — APPROVED** [review]

**Comment by manoelcampos** [comment]
Hey @sun-lingyu 
I'm really sorry for the very late reply.
I was away from the project for a few months but I'm back.

After reviewing your PR, I realized there were lots of issues with time attributes scattered across multiple classes such as Clodlet, CloudletExecution, and CloudletDatacenterExecution. There was a TODO comment in this last class, explaining some of the issues which were inherited from CloudSim. Those duplicated attributes were causing data inconsistency and were the source of many bugs.

Since you were dealing with a new time attribute that was suffering from those issues, I had to fix them.
Check the last commit comment for more details. 
Thanks for you initial work.

**Pull request merged**